### PR TITLE
Option to create a self-contained output file in "gsctl create kubeconfig"

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/gsctl/issues/113, based on https://github.com/giantswarm/gsctl/pull/161

This change adds the `--self-contained` flag to enable the creation of a dedicated kubectl config file containing CA certificate, client certificate and client key as inline data.

Example usage:

```nohighlight
$ gsctl create kubeconfig --ttl 1 -c fz8zh --self-contained ./self-contained.yaml
Creating new key pair…
New key pair created with ID 37f03cd6a… and expiry of 24 hours
Self-contained kubectl config file written to: ./self-contained.yaml

To make use of this file, run:

    export KUBECONFIG=./self-contained.yaml
    kubectl cluster-info

```

Contents of `self-contained.yaml` (redacted):

```yaml
apiVersion: v1
kind: Config
clusters:
- name: giantswarm-fz8zh
  cluster:
    server: https://api.fz8zh.g8s.geckon.gridscale.kvm.gigantic.io
    certificate-authority-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FU...
users:
- name: giantswarm-fz8zh-user
  user:
    client-certificate-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tL...
    client-key-data: LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLR...
contexts:
- name: giantswarm-fz8zh
  context:
    cluster: giantswarm-fz8zh
    user: giantswarm-fz8zh-user
current-context: giantswarm-fz8zh
preferences: {}
```

#### Caution

Existing output files are currently overwritten without a warning or confirmation.

### CODE REVIEW

**Comment by marians** [comment]
Binary for Mac OS:

[gsctl.zip](https://github.com/giantswarm/gsctl/files/1521521/gsctl.zip)

**Comment by webwurst** [comment]
Looks good to me! Just for comparison, a config created by `kubeadm` on installation. Nut sure if it's worth to align.
```yaml
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: __data__
    server: https://__address__:6443
  name: kubernetes
contexts:
- context:
    cluster: kubernetes
    user: kubernetes-admin
  name: kubernetes-admin@kubernetes
current-context: kubernetes-admin@kubernetes
kind: Config
preferences: {}
users:
- name: kubernetes-admin
  user:
    client-certificate-data: __data__
    client-key-data: __data__
```

**Comment by marians** [comment]
Thanks @webwurst! In terms of structure, this is exactly what we generate here.

**Comment by nexeck** [comment]
Works for me :)

**Comment by marians** [comment]
Update the usage info:

```
Creates or modifies kubectl configuration to access your Giant Swarm
Kubernetes cluster

By executing this command, you create a new key pair for your cluster to
authenticate with as a kubectl user.

By default, your kubectl config is modified to add user, cluster, and context
entries. The config file is assumed to be in $HOME/.kube/config. If set, the
path from the $KUBECONFIG environment variable is used. Certificate files are
stored in the "certs" subfolder of the gsctl config directory. See 'gsctl info'.

Alternatively, the --self-contained <path> flag can be used to create a new
config file with included certificates.

Examples:

  gsctl create kubeconfig -c my0c3

  gsctl create kubeconfig -c my0c3 --self-contained ./kubeconfig.yaml

  gsctl create kubeconfig -c my0c3 --ttl 1 -d "Short lived key pair"

  gsctl create kubeconfig -c my0c3 --certificate-organizations system:masters

Usage:
  gsctl create kubeconfig [flags]

Flags:
      --certificate-organizations string   A comma separated list of organizations for the issued certificates 'O' fields.
  -c, --cluster string                     ID of the cluster
      --cn-prefix string                   The common name prefix for the issued certificates 'CN' field.
  -d, --description string                 Description for the key pair
      --force                              If set, --self-contained will overwrite existing files without interafctive confirmation.
  -h, --help                               help for kubeconfig
      --self-contained string              Create a self-contained kubectl config with embedded credentials and write it to this path.
      --ttl int                            Duration until expiry of the created key pair in days (default 30)

Global Flags:
...
```

File: commands/create_kubeconfig.go [hunk 1]
```diff
@@ -75,13 +110,64 @@ type createKubeconfigResult struct {
...
+	User    string `yaml:"user"`
 }
 
 func init() {
 	CreateKubeconfigCommand.Flags().StringVarP(&cmdClusterID, "cluster", "c", "", "ID of the cluster")
 	CreateKubeconfigCommand.Flags().StringVarP(&cmdDescription, "description", "d", "", "Description for the key pair")
 	CreateKubeconfigCommand.Flags().StringVarP(&cmdCNPrefix, "cn-prefix", "", "", "The common name prefix for the issued certificates 'CN' field.")
+	CreateKubeconfigCommand.Flags().StringVarP(&cmdKubeconfigSelfContained, "self-contained", "", "", "Create a self-contained kubectl config with embedded credentials and write it to this path.")
 	CreateKubeconfigCommand.Flags().StringVarP(&cmdCertificateOrganizations, "certificate-organizations", "", "", "A comma separated list of organizations for the issued certificates 'O' fields.")
+	CreateKubeconfigCommand.Flags().BoolVarP(&cmdForce, "force", "", false, "If set, --self-contained will overwrite existing files without interafctive confirmation.")
```
Comment (nexeck) [inline commands/create_kubeconfig.go]: interafctive  > interactive

Comment (marians) [reply commands/create_kubeconfig.go]: Thanks!

**Pull request merged**